source/Tests/Testing/Integration/IntegrationTestForwardCancellationTokenToInvocationsAnalyzerFixture.cs

```diff
@@ -0,0 +1,426 @@
+    Task MethodAsync(string a, string b, CancellationToken ct) => Task.CompletedTask;
+}
+";
+            await Verify.VerifyAnalyzerAsync(container.WithTestingTypes());
+        }
+        
+        [TestCase]
+        public async Task No_Diagnostic_OverloadShouldHaveExactParameterTypePlusCancellationToken()
+        {
+            const string container = @"
+public class TestClass : IntegrationTest
+{
+    async Task M()
+    {
+        await {|#0:MethodAsync|}(""any"");
+    }
+
+    Task MethodAsync(string a) => Task.CompletedTask;
+    Task MethodAsync(int a, CancellationToken ct) => Task.CompletedTask;
+}
+";
+            await Verify.VerifyAnalyzerAsync(container.WithTestingTypes());
+        }
+        
+        [TestCase]
+        public async Task No_Diagnostic_TokenForwardedWithUnorderedNamedParameter()
+        {
+            const string container = @"
+public class TestClass : IntegrationTest
+{
+    async Task M()
+    {
+        await {|#0:MethodAsync|}(ct: CancellationToken, a: ""any"");
+    }
+
+    Task MethodAsync(string a) => Task.CompletedTask;
+    Task MethodAsync(string a, CancellationToken ct) => Task.CompletedTask;
+}
+";
+            await Verify.VerifyAnalyzerAsync(container.WithTestingTypes());
+        }
+        
+        [TestCase]
+        public async Task No_Diagnostic_ExtensionMethodTakesTokenAsync()
+        {
+            const string container = @"
+public class TestClass : IntegrationTest
+{
+    async Task M()
+    {
+        await {|#0:new MyClass()|}.MyMethod(""any"");
+    }
+}
+public class MyClass
+{
+    public Task MyMethod(string a) => Task.CompletedTask;
+}
+public static class Extensions
+{
+    public static Task MyMethod(this MyClass mc, CancellationToken c) => Task.CompletedTask;
+}
+";
+            await Verify.VerifyAnalyzerAsync(container.WithTestingTypes());
+        }
+        
+        [TestCase]
+        public async Task No_Diagnostic_StaticMethodBoundary()
+        {
+            const string container = @"
+public class TestClass : IntegrationTest
+{
+    public static async Task M()
+    {
+        await InnerMethod();                
+
+        Task InnerMethod(CancellationToken ct = default) => Task.CompletedTask;
+    }
+}
+";
+            await Verify.VerifyAnalyzerAsync(container.WithTestingTypes());
+        }
+        
+        [TestCase]
+        public async Task No_Diagnostic_OverloadReturnTypeDiffers()
+        {
+            const string container = @"
+public class TestClass : IntegrationTest
+{
+    async Task M()
+    {
+        await {|#0:Method|}(a: ""any"");
+    }
+
+    Task Method(string a) => Task.CompletedTask;
+    int Method(string a, CancellationToken ct) { throw new NotImplementedException(); }
+}
+";
+            await Verify.VerifyAnalyzerAsync(container.WithTestingTypes());
+        }
+        
+        [TestCase]
+        public async Task Diagnostic_MethodWithCancellationTokenOverload()
+        {
+            var container = @"
```

Comment:
nit - inconsistency + this should probably be a `const` anyway. All the strings in above tests are `const`, but everything below is using `var`.

Reply:
It's all consistent now with `const string container`